src/NHibernate/Transform/IResultTransformer.cs

```diff
@@ -16,13 +16,88 @@ public interface IResultTransformer
 		/// <param name="tuple"></param>
 		/// <param name="aliases"></param>
 		/// <returns></returns>
+		// Since v5.3
+		[Obsolete("Use TransformTuple extension method with parameterValues parameter instead.")]
 		object TransformTuple(object[] tuple, string[] aliases);
 
 		/// <summary>
 		/// 
 		/// </summary>
 		/// <param name="collection"></param>
 		/// <returns></returns>
+		// Since v5.3
+		[Obsolete("Use TransformList extension method with parameterValues parameter instead.")]
 		IList TransformList(IList collection);
 	}
-}
\ No newline at end of file
+
+	// TODO 6.0: Move to IResultTransformer
+	internal interface IResultTransformerExtended : IResultTransformer
```

Comment:
This `IResultTransformer` interface is quite an ugly thing and this extension doesn't make it any better... Can you think of any way to avoid this interface changes? Why do we need to have `parameterValues` as parameters? Why can't it be internal state of LINQ `ResultTransfomer` class? IMHO it's better to recreate it on each query execution if it avoids interface changes.  Is it really needed in any other place?

Reply:
> Can you think of any way to avoid this interface changes?

Maybe by creating a new instance of `ResultTransformer` each time a query is executed and pass `parameterValues` in the constructor instead. I will try if it is possible.